src/tests/scenarios/test_iso_yielder_setting.cairo

```diff
@@ -0,0 +1,731 @@
+    assert(claimed == 0, 'Wrong claimed: should be 0');
+
+
+    // At t = Sep 01 2023 07:00:00 GMT+0000 (1 month after first price at 22)
+    set_block_timestamp(1693551600);
+    // Claimable is 8781856 * 10 (UNIT_PRICE)
+    let claimable = yielder.get_claimable_of(signers.anyone);
+    assert(claimable == 8781856, 'Wrong claimable:need 8781856');
+    // Claimed is 0
+    let claimed = yielder.get_claimed_of(signers.anyone);
+    assert(claimed == 0, 'Wrong claimed');
+}
+
+
+#[test]
+#[available_gas(400_000_000)]
+fn deposit_and_withdraw_value_in_yielder() {
+    let (signers, contracts) = setup(PRICE);
+    // Instantiate contracts
+    let farmer = IFarmDispatcher { contract_address: contracts.yielder };
+    let yielder = IYieldDispatcher { contract_address: contracts.yielder };
+    let minter = IMinterDispatcher { contract_address: contracts.project };
+    let project = IProjectDispatcher { contract_address: contracts.project };
+    let absorber = IAbsorberDispatcher { contract_address: contracts.project };
+    let erc3525 = IERC3525Dispatcher { contract_address: contracts.project };
+    let erc20 = IERC20Dispatcher { contract_address: contracts.erc20 };
+
+    // Prank caller as owner
+    set_contract_address(signers.owner);
+
+    // Grant minter rights to owner, mint 1 token to anyone and revoke rights
+    minter.add_minter(SLOT, signers.owner);
+    let token_id = project.mint(signers.anyone, SLOT, VALUE);
+    minter.revoke_minter(SLOT, signers.owner);
+
+    // Setup project value
+    let project_value = erc3525.total_value(SLOT);
+    absorber.set_project_value(SLOT, project_value);
+
+    // Prank caller as anyone
+    set_contract_address(signers.anyone);
+
+    // At t = 0
+    set_block_timestamp(0);
+    // Anyone deposits value 100_000_000 in yielder
+    farmer.deposit(token_id, VALUE);
+    let deposited = farmer.get_deposited_of(signers.anyone);
+    assert(deposited == VALUE, 'Wrong strat deposited');
+    
+    //check balance of anyone need to be reviewed 
+    let balance = erc20.balance_of(signers.anyone);
+    assert(balance == 0, 'Wrong balce of anyone shld be 0');
+
+
+    // At t = Aug 01 2023 07:00:00 GMT+0000 (3h before first price other than 0)
+    set_block_timestamp(1690873200); 
+    // Then `get_claimable_of` should be  0
+    let claimable = yielder.get_claimable_of(signers.anyone);
+    assert(claimable == 0, 'Wrong claimable: should be 0');
+
+
+    // At t = Sep 01 2023 07:00:00 GMT+0000 (1 month after first price at 22)
+    set_block_timestamp(1693551600);
+
+    // Then `get_claimable_of` should be different from 0
+    let claimable = yielder.get_claimable_of(signers.anyone);
+    assert(claimable != 0, 'Wrong claimable should not be 0');
+
+    // withdraw VALUE from yielder 
+    farmer.withdraw_to_token(token_id, VALUE);
+    // balance should be egal to value deposited
+    //get value on token_id for anyone
+    let value = erc3525.total_value(SLOT);
+    value.print();
```

Comment:
Use erc3525.value_of(token_id) to compare the withdrawn value with the expected one.

Reply:
done